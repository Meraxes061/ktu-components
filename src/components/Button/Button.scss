@import '../../styles/style.scss';

.c-button {
    background: transparent;
    border: 1px solid transparent;
    border-radius: 4px;
    cursor: pointer;
    font-family: $font-family;

    &--primary {
        background-color: $primary-color;
        color: white;
        outline: none;
    }

    &--secondary {
        background-color: $secondary-color;
        color: white;
        outline: none;
    }

    &--small {
        padding: 4px 8px;
        font-size: $font-small;
    }

    &--medium {
        padding: 6px 12px;
        font-size: $font-medium;
    }

    &--large {
        padding: 9px 18px;
        font-size: $font-large;
    }

    &--block {
        display: block;
        width: 100%;
    }
}