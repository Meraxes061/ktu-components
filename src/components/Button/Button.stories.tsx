import React from 'react'
import { storiesOf } from '@storybook/react'
import Button from './Button'

storiesOf('Button', module).add('Default', () => (
    <div style={{ fontFamily: 'Segoe UI' }}>
        <h1 style={{ margin: 8 }}>Default Button</h1>
        <Button>Default</Button>
        <h2 style={{ margin: '16px 8px ' }}>Sizes</h2>
        <div style={{ display: 'flex', alignItems: 'baseline' }}>
            <div style={{ marginRight: 16 }}>
                <Button size='small'>Small</Button>
            </div>
            <div style={{ marginRight: 16 }}>
                <Button size='medium'>Medium</Button>
            </div>
            <div style={{ marginRight: 16 }}>
                <Button size='large'>Large</Button>
            </div>
        </div>
        <h2 style={{ margin: '16px 8px ' }}>Block</h2>
        <Button block>Block</Button>
    </div>
))
storiesOf('Button', module).add('Primary', () => (
    <div style={{ fontFamily: 'Segoe UI' }}>
        <h1 style={{ margin: 8 }}>Primary Button</h1>
        <Button theme='primary'>Primary</Button>
        <h2 style={{ margin: '16px 8px ' }}>Sizes</h2>
        <div style={{ display: 'flex', alignItems: 'baseline' }}>
            <div style={{ marginRight: 16 }}>
                <Button theme='primary' size='small'>Small</Button>
            </div>
            <div style={{ marginRight: 16 }}>
                <Button theme='primary' size='medium'>Medium</Button>
            </div>
            <div style={{ marginRight: 16 }}>
                <Button theme='primary' size='large'>Large</Button>
            </div>
        </div>
        <h2 style={{ margin: '16px 8px ' }}>Block</h2>
        <Button block theme='primary'>Block</Button>
    </div>
))
storiesOf('Button', module).add('Secondary', () => (
    <div style={{ fontFamily: 'Segoe UI' }}>
        <h1 style={{ margin: 8 }}>Secondary Button</h1>
        <Button theme='secondary'>Secondary</Button>
        <h2 style={{ margin: '16px 8px ' }}>Sizes</h2>
        <div style={{ display: 'flex', alignItems: 'baseline' }}>
            <div style={{ marginRight: 16 }}>
                <Button theme='secondary' size='small'>Small</Button>
            </div>
            <div style={{ marginRight: 16 }}>
                <Button theme='secondary' size='medium'>Medium</Button>
            </div>
            <div style={{ marginRight: 16 }}>
                <Button theme='secondary' size='large'>Large</Button>
            </div>
        </div>
        <h2 style={{ margin: '16px 8px ' }}>Block</h2>
        <Button block theme='secondary'>Block</Button>
    </div>
))