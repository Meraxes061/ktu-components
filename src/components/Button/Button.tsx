import React, { FunctionComponent, ReactNode } from 'react'
import cx from 'classnames'

import './Button.scss'

type theme = 'default' | 'primary' | 'secondary'
type size = 'small' | 'medium' | 'large'

type Props = JSX.IntrinsicElements['button'] & {
    theme?: theme,
    size?: size,
    block?: boolean,
    children: ReactNode,
}
const Button: FunctionComponent<Props> = ({ children, theme = 'default', size = 'medium', block = false, ...rest }) => {
    const className = cx(
        'c-button',
        {
            [`c-button--${theme}`]: theme,
            [`c-button--${size}`]: size,
            [`c-button--block`]: block,
        }
    )
    return (
        <button className={className} {...rest}>{children}</button>
    )
}

export default Button;