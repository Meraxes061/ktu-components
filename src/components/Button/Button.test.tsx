import React from 'react'
import { shallow, mount } from 'enzyme'
import Button from './Button'

describe('Button', () => {
    it('renders correctly button which is unspecified any property', () => {
        const wrapper = shallow(<Button>No property variation</Button>)
        expect(wrapper.find('button').hasClass('c-button c-button--default c-button--medium')).toEqual(true)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly button which is size is small', () => {
        const wrapper = shallow(<Button size='small'>Small variation</Button>)
        expect(wrapper.find('button').hasClass('c-button c-button--default c-button--small')).toEqual(true)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly button which is size is medium', () => {
        const wrapper = shallow(<Button size='medium'>Medium variation</Button>)
        expect(wrapper.find('button').hasClass('c-button c-button--default c-button--medium')).toEqual(true)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly button which is size is large', () => {
        const wrapper = shallow(<Button size='large'>Large variation</Button>)
        expect(wrapper.find('button').hasClass('c-button c-button--default c-button--large')).toEqual(true)
        expect(wrapper).toMatchSnapshot()
    })

    it('renders correctly button which is theme is default', () => {
        const wrapper = shallow(<Button>Default variation</Button>)
        expect(wrapper.find('button').hasClass('c-button c-button--default c-button--medium')).toEqual(true)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly button which is theme is primary', () => {
        const wrapper = shallow(<Button theme='primary'>Primary variation</Button>)
        expect(wrapper.find('button').hasClass('c-button c-button--primary c-button--medium')).toEqual(true)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly button which is theme is secondary', () => {
        const wrapper = shallow(<Button theme='secondary'>Secondary variation</Button>)
        expect(wrapper.find('button').hasClass('c-button c-button--secondary c-button--medium')).toEqual(true)
        expect(wrapper).toMatchSnapshot()
    })
    it('simulate click event correctly', () => {
        const mockFn = jest.fn()
        const wrapper = mount(<Button onClick={mockFn}>Simulating Click</Button>)
        wrapper.find('button').simulate('click')
        expect(mockFn).toHaveBeenCalled()
    })

})