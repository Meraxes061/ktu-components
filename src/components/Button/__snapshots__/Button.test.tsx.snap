// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly button which is size is large 1`] = `
<button
  className="c-button c-button--default c-button--large"
>
  Large variation
</button>
`;

exports[`Button renders correctly button which is size is medium 1`] = `
<button
  className="c-button c-button--default c-button--medium"
>
  Medium variation
</button>
`;

exports[`Button renders correctly button which is size is small 1`] = `
<button
  className="c-button c-button--default c-button--small"
>
  Small variation
</button>
`;

exports[`Button renders correctly button which is theme is default 1`] = `
<button
  className="c-button c-button--default c-button--medium"
>
  Default variation
</button>
`;

exports[`Button renders correctly button which is theme is primary 1`] = `
<button
  className="c-button c-button--primary c-button--medium"
>
  Primary variation
</button>
`;

exports[`Button renders correctly button which is theme is secondary 1`] = `
<button
  className="c-button c-button--secondary c-button--medium"
>
  Secondary variation
</button>
`;

exports[`Button renders correctly button which is unspecified any property 1`] = `
<button
  className="c-button c-button--default c-button--medium"
>
  No property variation
</button>
`;
